lib/SPIRV/libSPIRV/SPIRVAsm.h

```diff
@@ -0,0 +1,155 @@
+//===- SPIRVAsm.h -                                           --*- C++ -*-===//
+//
+//                     The LLVM/SPIRV Translator
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+/// \file
+///
+/// This file defines the inline assembler entries defined in SPIRV spec with op
+/// codes.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef SPIRV_LIBSPIRV_SPIRVASM_H
+#define SPIRV_LIBSPIRV_SPIRVASM_H
+
+#include "SPIRVEntry.h"
+#include "SPIRVInstruction.h"
+#include "SPIRVStream.h"
+
+#include <cassert>
+#include <map>
+#include <tuple>
+#include <vector>
+
+namespace SPIRV {
+
+class SPIRVAsmTargetINTEL : public SPIRVEntry {
+public:
+  static const SPIRVWord FixedWC = 2;
+  static const Op OC = OpAsmTargetINTEL;
+  // Complete constructor
+  SPIRVAsmTargetINTEL(SPIRVModule *M, SPIRVId TheId,
+                      const std::string &TheTarget)
+      : SPIRVEntry(M, FixedWC + getSizeInWords(TheTarget), OC, TheId),
+        Target(TheTarget) {
+    validate();
+  }
+  // Incomplete constructor
+  SPIRVAsmTargetINTEL() : SPIRVEntry(OC) {}
+  SPIRVCapVec getRequiredCapability() const override {
+    return getVec(CapabilityAsmINTEL);
+  }
+  SPIRVExtSet getRequiredExtensions() const override {
+    return getSet(ExtensionID::SPV_INTEL_inline_assembly);
+  }
+  const std::string &getTarget() const { return Target; }
+
+protected:
+  void validate() const override {
+    SPIRVEntry::validate();
+    assert(WordCount > FixedWC);
+    assert(OpCode == OC);
+  }
+  _SPIRV_DEF_ENCDEC2(Id, Target)
+  std::string Target;
+};
+
+class SPIRVAsmINTEL : public SPIRVValue {
+public:
+  static const SPIRVWord FixedWC = 5;
```

Comment:
According to the spec `OpAsmINTEL` must have at least 7 operands

Reply:
FixedWC is not a total word count.